examples/architectureComponentsExample/src/main/java/io/realm/examples/arch/livemodel/LiveRealmObject.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.examples.arch.livemodel;
+
+import android.arch.lifecycle.LiveData;
+import android.support.annotation.MainThread;
+
+import io.realm.ObjectChangeSet;
+import io.realm.RealmModel;
+import io.realm.RealmObject;
+import io.realm.RealmObjectChangeListener;
+
+
+public class LiveRealmObject<T extends RealmModel> extends LiveData<T> {
```

Comment:
I don't think either approach is wrong, to be honest. Your approach put all of the responsibility in the ViewModel vs. this one where it lives in the LiveData object instead.

For small ViewModels, your approach seems fine to me, but I would be worried about how well it scales since you need to manually manage all the listeners. Having that inside the LiveData object means you don't need to write that code manually each time.